proposals/materialx-versioning/README.md

```diff
@@ -0,0 +1,130 @@
+
+## Proposal to Version MaterialX data in OpenUSD
+### Overview
+We propose adding an applied API schema `MaterialXInfoAPI` (in the spirit of the 
+["Revise use of Layer Metadata in USD"](https://github.com/PixarAnimationStudios/OpenUSD-proposals/blob/main/proposals/revise_use_of_layer_metadata/README.md)) that can record the MaterialX library version. We keep the name of the 
+applied API schema a little more generalized to leave the door open for additional MaterialX document information to be
+added to the schema at a later date. We deliberately decide to only propose adding the MaterialX library version to the 
+schema to keep the proposal as uncontroversial as possible to add rapid adoption.
+
+The `MaterialXInfoAPI` will be applied to the UsdShade `Material` primitive. This allows for referencing the material 
+directly from a `.mtlx` file and still retaining the MaterialX library version when the scene is composed. i.e.
+```
+def "Material" MyMaterial (
+    references=@myMaterials.mtlx@</MaterialX/Materials/MyMaterial>
+)
+{}
+```
+
+Any UsdShade `Material` primitives that do not have the applied API would be assumed to be version 1.38, thus providing a 
+path forward to support existing OpenUSD files that are not using this applied API schema.
+
+With the MaterialX library version recorded in the OpenUSD stage, `HdMtlx` would now have access to an appropriate 
+version number to use when re-constructing the MaterialX document. This will then allow the existing MaterialX upgrade 
+functionality to be used.
+
+### UsdMtlx Implementation changes
+
+The only change required in `UsdMtlx` would be to apply the applied API schema to the `Material` primitive, and set the
+MaterialX library version.  This version number would come directly from the `.mtlx` file that was being transcoded to 
+OpenUSD.
+
+### HdMtlx Implementation changes.
+
+The `HdMtlx` library will also need to be changed to read the MaterialX library version from the `Material` primitive, 
+and author this version number in the reconstructed MaterialX document.
+
+If the version is not present it will be assumed to be v1.38. 
+
+## Out of Scope
+We do not propose any sort of validation regarding the version other than its presence, and then falling back
+to the default if missing. 
+
+Specifically we do not propose attempting to validate that all opinions that make up a composed
+`Material` primitive were authored with the same MaterialX version, even though this composition could lead to an
+invalid MaterialX material, for a number of reasons.
+* The expense of the validation, in some circumstances, could be too high to consider.
+* It is possible that a `Material` primitive may still contain a valid MaterialX material even if composed of differing
+  MaterialX library versions.
+* Reporting the error during the evaluation of Hydra may be later in a pipeline than desired.
+* Some implementations might not even use the `HdMtlx` library to extract the MaterialX material.
```

Comment:
You are now able to write a UsdValidator test that can perform such a validation independently of stage-construction (UsdMtlx) or rendering.